packages/elements/src/internals/machines/sign-in/verification.machine.ts

```diff
@@ -236,7 +241,10 @@ const SignInVerificationMachine = setup({
         input: ({ context }) => ({
           parent: context.parent,
           resendable: context.resendable,
-          params: context.currentFactor as PrepareFirstFactorParams,
+          params: {
+            ...context.currentFactor,
+            redirectUrl: `${window.location.origin}${context.basePath}${MAGIC_LINK_VERIFY_PATH_ROUTE}`,
```

Comment:
It might make sense for this URL to be built somewhere else. In clerk-js we have a buildURL helper that handles appending the origin based on the routing type.